src/actions/Bluetooth/SensorUpdateActions.js

```diff
@@ -92,19 +95,36 @@ const startSetLogInterval = ({ macAddress, logInterval = 300 }) => async (dispat
   return result;
 };
 
-const updateSensor = sensor => async dispatch => {
-  if (!isValidMacAddress(sensor.macAddress)) {
+const updateNewSensor = sensor => async dispatch => {
+  await dispatch(updateSensor(sensor));
+  await dispatch(SensorUpdateActions.startSensorDisableButton(sensor.macAddress));
+};
+
+const updateSensor = sensor => async (dispatch, getState) => {
+  const { macAddress } = sensor;
+
+  if (!isValidMacAddress(macAddress)) {
     // prevent errors if the sensor is not able to be contacted
     return;
   }
 
+  const oldLogInterval = UIDatabase.get('Sensor', macAddress, 'macAddress')?.logInterval;
+  if (oldLogInterval && sensor.logInterval === oldLogInterval) {
+    // No updates required if no change
+    return;
+  }
+
+  if (selectIsSyncingTemps(getState())) {
```

Comment:
Sorry! Only just thought of this.. should it be `selectIsDownloading(getState(), macAddress)`?
Does it matter if another sensor is downloading?

Reply:
I've had really mixed results when another sensor is downloading (I tried the above)... on one hand it's really annoying to be blocked when another sensor is downloading, but on the other I've been seeing sensor disconnect problems if another one is downloading... maybe that is something worth digging into further 🕳️ ... hmm

Reply:
Good enough for me! thanks.